File: fixtures/theme-audit/sample.ts
// Hearth fixture: TypeScript semantic coverage
type ID = string | number

enum BuildMode {
  Dev = 'dev',
  Prod = 'prod'
}

interface UserProfile {
  readonly id: ID
  name: string
  createdAt: Date
}

class Repository<T extends UserProfile> {
  constructor(private readonly records: Map<ID, T>) {}

  findById(id: ID): T | undefined {
    return this.records.get(id)
  }
}

const baseUrl = 'https://api.hearthcode.dev'
const timeoutMs = 5_000
const strict = true
const idPattern = /^[a-z0-9_-]+$/i

File: fixtures/theme-audit/sample.py
# Hearth fixture: Python semantic coverage
from dataclasses import dataclass
from enum import Enum
from typing import Optional

API_ROOT = "https://api.hearthcode.dev"
DEFAULT_TIMEOUT = 30.0


class LogLevel(Enum):
    INFO = "info"
    WARN = "warn"
    ERROR = "error"


@dataclass
class Session:
    token: str
    user_id: int
    expires_at: Optional[str] = None


def load_session(raw: dict[str, object]) -> Optional[Session]:
    if "token" not in raw or "user_id" not in raw:
        return None
    return Session(token=str(raw["token"]), user_id=int(raw["user_id"]))
